src/hotspot/share/classfile/stackMapFrame.hpp

```diff
@@ -136,6 +155,64 @@ class StackMapFrame : public ResourceObj {
   inline u2 max_stack() const                 { return _max_stack; }
   inline bool flag_this_uninit() const        { return _flags & FLAG_THIS_UNINIT; }
 
+  AssertUnsetFieldTable* assert_unset_fields() const {
+    return _assert_unset_fields;
+  }
+
+  void set_assert_unset_fields(AssertUnsetFieldTable* table) {
+    _assert_unset_fields = table;
+  }
+
+  bool satisfy_unset_field(Symbol* name, Symbol* signature) {
+    NameAndSig dummy_field(name, signature);
+
+    if (_assert_unset_fields->contains(dummy_field)) {
+      NameAndSig* field = _assert_unset_fields->get(dummy_field);
+      field->_satisfied = true;
+      return true;
+    }
+    return false;
+  }
+
+  bool is_unset_fields_satisfied() {
```

Comment:
Also add a comment when this is called - ie. when you get to the super() call?

Maybe also this one could be verify_unset_fields_are_satisfied().